tobac/tests/test_segmentation.py

```diff
@@ -112,3 +169,198 @@ def test_segmentation_timestep_level():
+        field_in=test_data_iris,
+        features_in=test_feature_ds,
+        dxy=test_dxy,
+        threshold=1.5,
+        seed_3D_flag="box",
+        seed_3D_size=seed_3D_size,
+    )
+
+    second_point_seg = out_seg_mask.core_data()[
+        int(test_vdim_pt_1 + shift_pts[0]),
+        int(test_hdim_1_pt_1 + shift_pts[1]),
+        int(test_hdim_2_pt_1 + shift_pts[2]),
+    ]
+    # We really only need to check the center point here for this test.
+    seg_point_overlaps = second_point_seg == 1
+    assert seg_point_overlaps == expected_both_segmented
+
+
+@pytest.mark.parametrize(
+    "test_dset_size, vertical_axis_num, "
+    "vertical_coord_name,"
+    " vertical_coord_opt, expected_raise",
+    [
+        ((20, 30, 40), 0, "altitude", "auto", False),
+        ((20, 30, 40), 1, "altitude", "auto", False),
+        ((20, 30, 40), 2, "altitude", "auto", False),
+        ((20, 30, 40), 0, "air_pressure", "air_pressure", False),
+        ((20, 30, 40), 0, "air_pressure", "auto", True),
+        ((20, 30, 40), 0, "model_level_number", "auto", False),
+        ((20, 30, 40), 0, "altitude", "auto", False),
+        ((20, 30, 40), 0, "geopotential_height", "auto", False),
+    ],
+)
+def test_different_z_axes(
+    test_dset_size,
+    vertical_axis_num,
+    vertical_coord_name,
+    vertical_coord_opt,
+    expected_raise,
+):
+    """Tests ```tobac.segmentation.segmentation_timestep```
+    Tests:
+    The output is the same no matter what order we have axes in.
+    A ValueError is raised if an invalid vertical coordinate is
+    passed in
+
+    Parameters
+    ----------
+    test_dset_size: tuple(int, int, int)
+        Size of the test dataset
+    vertical_axis_num: int (0-2, inclusive)
+        Which axis in test_dset_size is the vertical axis
+    vertical_coord_name: str
+        Name of the vertical coordinate.
+    vertical_coord_opt: str
+        What to pass in as the vertical coordinate option to segmentation_timestep
+    expected_raise: bool
+        True if we expect a ValueError to be raised, false otherwise
+    """
+    import numpy as np
+
+    # First, just check that input and output shapes are the same.
+    test_dxy = 1000
+    test_vdim_pt_1 = 8
+    test_hdim_1_pt_1 = 12
+    test_hdim_2_pt_1 = 12
+    test_data = np.zeros(test_dset_size)
+    common_dset_opts = {
+        "in_arr": test_data,
+        "data_type": "iris",
+        "z_dim_name": vertical_coord_name,
+    }
+    if vertical_axis_num == 0:
+        test_data_iris = testing.make_dataset_from_arr(
+            z_dim_num=0, y_dim_num=1, x_dim_num=2, **common_dset_opts
+        )
+    elif vertical_axis_num == 1:
+        test_data_iris = testing.make_dataset_from_arr(
+            z_dim_num=1, y_dim_num=0, x_dim_num=1, **common_dset_opts
+        )
+    elif vertical_axis_num == 2:
+        test_data_iris = testing.make_dataset_from_arr(
+            z_dim_num=1, y_dim_num=0, x_dim_num=1, **common_dset_opts
```

Comment:
`zdim_num = 2` ; if this is changed to the correct axis number the test fails, meaning that the bug related to the transposing in `segmentation_timestep()`is successfully captured

Reply:
I've fixed this, it did catch the bug, and I've now fixed the bug.